### DISCUSSION TITLE
action VS autoActions best practice

### DISCUSSION THREAD

**Discussion by RoCat** [mobxjs/mobx#3531] [Q&A] [answered]
Hi,

I have a simplified mobx state:
```
export class DateMobx {
  dateTo: string | null = null;
  
  constructor() {
    this.init();
    makeAutoObservable(this);
  }
  setDateTo(value: string) {
    this.dateTo = value;
  }
}
```

And elsewhere in app I call dateMobx.setDateTo in an autorun. I get a warning because autoActions are not untracked.
What is the best practice to remove that warning? 

using a setter `set setDateTo ...`and call it with ` dateMobx.setDateTo = 'XXX'`(seems weird to me)
or calling the autoAction into a `runInAction`
or calling it into a `untracked` (seems lighter than autorun ?)
Or another solution I did,'t mentioned ?

**Comment by urugator** [comment, marked as answer]
Mutating state as part of `autorun/reaction` is discouraged:
https://mobx.js.org/reactions.html#use-reactions-sparingly
If you still want to do so I would go with
```javascript
autorun(() => {
  // yada yada
  runInAction(() => {
     dateMobx.setDateTo('XXX')
  })
})
// or (the effect is implicitly action)
reaction(() => `${foo.year}-${foo.month}-${foo.day}`, dateString => dateMobx.setDateTo(dateString))
```
> Or another solution I didn't mentioned ?

`makeAutoObservable(this, { setDateTo: action });`

**Reply by RoCat** [reply]
Tanks for this clear  answer.
